File: Year 2/Semester 4/Programming and Projection Environments/Week 10/ProtobuffProjectJavaORM/Persistance/src/main/java/gRPCProject/persistance/repository/IEmployeeRepository.java
package gRPCProject.persistance.repository;

import gRPCProject.domain.Entity;

import java.io.Serializable;

public interface IEmployeeRepository <ID extends Serializable, E extends Entity<ID>> extends IRepository<ID, E> {
    E authenticateEmployee(String username, String password);
}

File: Year 2/Semester 4/Programming and Projection Environments/Week 6/ASaseaSaptamana/ASaseaSaptamana/repository/interfaces/IEmployeeRepository.cs
using ASaseaSaptamana.domain;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ASaseaSaptamana.repository.interfaces
{
    public interface IEmployeeRepository : IRepository<int, Employee<int>>
    {
        Employee<int> AuthenticateEmployee(string username, string password);
    }
}
